pkg/cidata/cidata.TEMPLATE.d/boot/04-persistent-data-volume.sh

```diff
@@ -86,6 +102,14 @@ if [ "$(awk '$2 == "/" {print $3}' /proc/mounts)" == "tmpfs" ]; then
 		if [ -d /mnt/data"${DIR}" ]; then
 			mkdir -p "${DIR}"
 			mount --bind /mnt/data"${DIR}" "${DIR}"
+			# Remount submounts on top of the new ${DIR}
```

Comment:
I was wondering why this is needed. Is it just because `mount -a` does not deal with mountpoints with spaces, or is there another reason too?

Either way, the `mount -a` should no longer be needed now.

Reply:
The script bind-mounts `/home` after mounting `/home/USER` and makes `/home/USER` invisible,  but `mount -a` still thinks `/home/USER` is properly mounted, so `mount -a` just skips `/home/USER`.

Reply:
Where is `/home/$USER` being remounted. You unmount it above, and then you bind-mount `/home` here. Why does `mount -a` think that `/home/$USER` is still mounted when it has been explicitly unmounted?

Note that I'm just asking to understand. This is obviously the right way to do it because `mount -a` does not handle the mount points with space.

Reply:
> You unmount it above,

Only for the first boot.